app/src/main/java/com/xmartlabs/scasas/doapp/ui/grouplist/GroupsListFragment.java

```diff
@@ -0,0 +1,474 @@
+        updateAll(groupList);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private Group getGroup(@NonNull String groupName) {
+    return Stream.of(groups)
+        .filter(group -> Objects.equals(groupName, group.getName()))
+        .findFirst()
+        .get();
+  }
+
+  private Group getGroup(@StringRes int groupName) {
+    String name = getResources().getString(groupName);
+    return getGroup(name);
+  }
+
+  private void insertTask(Task task, Group group) {
+    taskController.insertTask(task)
+        .subscribe(new SingleSubscriber<Task>() {
+          @Override
+          public void onSuccess(Task task) {
+            Toast.makeText(getContext(), R.string.task_added, Toast.LENGTH_SHORT).show();
+            updateItemCount(group.getName());
+            closeNewTaskView();
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+          }
+        });
+  }
+
+  private void updateItemCount(String groupName) {
+    int group = getStringRes(groupName);
+    if (group != 0) {
+      updateItemCount(group);
+    }
+  }
+
+  private void updateItemCount(@StringRes int groupName) {
+    switch (groupName) {
+      case R.string.shop:
+        getTaskCount(R.string.shop);
+        break;
+      case R.string.work:
+        getTaskCount(R.string.work);
+        break;
+      case R.string.health:
+        getTaskCount(R.string.health);
+        break;
+      case R.string.travel:
+        getTaskCount(R.string.travel);
+        break;
+      case R.string.bills:
+        getTaskCount(R.string.bills);
+        break;
+      case R.string.auto:
+        getTaskCount(R.string.auto);
+        break;
+      default:
+        updateAll(groups);
+    }
+  }
+
+  private void updateAll(List<Group> groups) {
+    Stream.of(groups).forEach(this::getTaskCount);
+  }
+
+  private void getTaskCount(@StringRes int groupName) {
+    Group group = getGroup(groupName);
+    taskController.getTasksCount(user, group).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long taskCount) {
+        int groupName = getStringRes(group.getName());
+        updateItemField(groupName, taskCount);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private void getTaskCount(Group group) {
+    taskController.getTasksCount(user, group).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long taskCount) {
+        int groupName = getStringRes(group.getName());
+        updateItemField(groupName, taskCount);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private String getTextFromStringRes(@StringRes int text) {
+    return getResources().getString(text);
+  }
+
+  private int getStringRes(String text) {
```

Comment:
Yep, It's totally ninja hahaha I suppose that it must be a better way but i didn't have enough time to look up for it..